x/mongo/driver/topology/topology.go

```diff
@@ -597,11 +584,22 @@ func (t *Topology) processSRVResults(parsedHosts []string) bool {
 		t.fsm.removeServerByAddr(addr)
 		t.publishServerClosedEvent(s.address)
 	}
+
+	// Now that we've removed all the hosts that disappeared from the SRV record, we need to add any
+	// new hosts added to the SRV record. Shuffle the list of added hosts so we add them in random
+	// order. Stop adding hosts whenever we reach srvMaxHosts.
+	random.Shuffle(len(diff.Added), func(i, j int) {
```

Comment:
Shouldn't we only `Shuffle` if `srvMaxHosts > 0`?

Reply:
The [Polling SRV records specification](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#querying-dns) indicates shuffling only if necessary:

> If srvMaxHosts is zero or greater than or equal to the number of hosts in the DNS result, the driver MUST populate the seedlist with all hosts.
>
> If srvMaxHosts is greater than zero and less than the number of hosts in the DNS result, the driver MUST randomly select that many hosts and use them to populate the seedlist. Drivers SHOULD use the Fisher-Yates shuffle for randomization.

A possible impact of always shuffling is a shuffled order of ServerOpening events.

I agree with the suggested change of checking if `srvMaxHosts == 0` to shuffle along with the condition `srvMaxHosts < len(diff.Added)`.

Reply:
Good idea, will update.